### ISSUE TITLE
multi condition QTL detection

### ISSUE THREAD

**Issue by dphansti** [single-cell-genetics/limix_qtl#36]
We have generated RNA-seq data for the same donors across two conditions.  We have called QTLs on each condition separately. But to increase power we would like to perform a joint analysis treating the two profiles as repeat measures. 

Our first goal is to just increase power to detect QTLs.
Our secondary goal is to find condition-specific (response) QTLs.

My understanding is that run_interaction_QTL_analysis.py can accomplish goal #2.  But I don't see a way to accomplish goal #1 with limix_qtl. run_QTL_analysis.py can identify QTLs separately for each condition but not jointly. Is that correct?  If it is possible to identify QTLs based on the joint analysis can you let us know how.

**Comment by Bonder-MJ** [comment]
Dear Doug,

You can use both samples per condition jointly in the model as you do for
the interaction. So both your use cases are actually possible. I would add
the a term in your covariates describing the condition though. The related
nature of the data is handelde internally by the sample mapping and
kinship/ random effect term.

Happy to help if further instructions are needed.

Best,
Marc

**Comment by Bonder-MJ** [comment]
Hi Doug,

I need to check the current interaction model that is currently online.

The latest version actually has both these betas returned. So you can get
the SNP effect directly from the interaction model also. However, if you
want that permutation p values from the latter that I don't return.
therefor I was suggesting a first scan with the normal qtl scanner. Which
would get you the eQTLS to look further into for interaction qtls. With
permutations the interactions take a long time.

Best,
Marc

**Comment by dphansti** [comment]
Thanks again for your quick response and for developing this code in the first place.

Yes, we initially called QTLs in each condition separately and then tested for conditional QTLs using a linear mixed model.  But reviewers have specifically asked us to perform the initial QTL calling in a joint fashion to improve power. We could use `run_QTL_analysis.py` but it isn't ideal since it wouldn't include the interaction term.

Based on the outputs we got using the current version of limix_qtl we think that `run_interaction_QTL_analysis.py` only outputs pvalues/betas for the interaction term.  But it is possible that we are misinterpreting the outputs or not running it correctly. If there is a newer version that includes pvalues/betas for the SNP, we would love to use it!

**Comment by Bonder-MJ** [comment]
Yeah indeed it wouldn't include the SNP interaction term. But the interactions are significantly slower as you can't fix the null model since it has to change with every SNP.  For speed I would run the normal code with the term as an covariate and do use the permutations for calibrations. 

I will however, also push the updated code that creates three sets of outputs, and an updated result extractor that properly coverts these.
